concrete/src/Install/Installer.php

```diff
@@ -177,4 +182,46 @@ private function getDefaultConnectionConfiguration()
 
         return $result;
     }
+
+    /**
+     * @param \Concrete\Core\Database\Connection\Connection $connection
+     * @param array $configuration
+     *
+     * @return \Concrete\Core\Database\Connection\Connection
+     */
+    private function checkUnicodeCharset(Connection $connection, array $configuration)
```

Comment:
The cyclomatic complexity of this PR is pretty high, especially with those ternaries. And without comments, it's really challenging to review.

Can you break this out into a few different methods with lower complexity? That will help with the maintainability, readability, and testability in the future.